daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SequenceParserBases.scala

```diff
@@ -44,6 +50,45 @@ abstract class SequenceParserBase(
   override lazy val childProcessors = childParsers
 }
 
+abstract class UnorderedSequenceParserBase(
+  srd: SequenceRuntimeData,
+  choiceParser: ChoiceParser)
+  extends SequenceParserBase(srd, Vector(choiceParser)) {
+
+  override protected def parse(pstate: PState) = {
+
+    var isDone = false
+
+    while (!isDone && (pstate.processorStatus eq Success)) {
```

Comment:
If this loop is being used in the separated case, then this can't work because it can't handle separatorPosition="postfix". It has no way to decide to parse a choice child followed by a postfix separator because it is assuming the sep parser has been melded into the choiceParser.

Reply:
While I'm not saying that I am doing things correctly, I do have tests with separatorPosition=prefix and postfix that all produce the expected results.  That being said, I'm sure I haven't tested every edge case.

I'll start working on making the changes to use OrderedSeparatedParser instead of the Unordered one.

Reply:
Just talked with Steve and I agree that this is wrong and probably wouldn't work for some negative test cases.  I'll add a few of those to verify that the new code is working correctly.